java/org/contikios/cooja/plugins/ScriptRunner.java

```diff
@@ -413,16 +375,49 @@ private void updateTitle() {
     frame.setTitle(title);
   }
 
+  /** Check if the script has been updated and offer the user to save the changes. */
+  private void checkForUpdatesAndSave() {
+    if (!codeEditorChanged) {
+      return;
+    }
+    String s1 = "Yes";
+    String s2 = "No";
+    Object[] options = {s1, s2};
+    if (JOptionPane.showOptionDialog(Cooja.getTopParentContainer(),
+            "Do you want to save the changes to " + linkedFile.getName() + "?",
+            "Save script changes", JOptionPane.YES_NO_OPTION,
+            JOptionPane.WARNING_MESSAGE, null, options, s1) == JOptionPane.YES_OPTION) {
+      saveAsAction.actionPerformed(null);
+    }
```

Comment:
I like getting a warning when there is a risk of losing modifications. It would have been nice to have the option to cancel, but I do not think that would be possible to add easily. Instead of canceling, what do you think about also having the option to save to a new file if someone wants to keep the modifications but do not want to overwrite the original file? Perhaps something for a future PR?

Reply:
Yeah, good ideas but not that easy to implement with the current structure, let's do that in a future PR.